sql/src/main/java/io/crate/expression/scalar/postgres/CurrentSettingFunction.java

```diff
@@ -0,0 +1,93 @@
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.postgres;
+
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.FunctionName;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.functions.params.Param;
+import io.crate.metadata.pgcatalog.PgCatalogSchemaInfo;
+import io.crate.metadata.settings.session.SessionSetting;
+import io.crate.metadata.settings.session.SessionSettingRegistry;
+import io.crate.types.DataType;
+
+import java.util.List;
+import java.util.Locale;
+
+public class CurrentSettingFunction extends Scalar<String, Object> {
+
+    private static final String NAME = "current_setting";
+    private static final FunctionName FQN = new FunctionName(PgCatalogSchemaInfo.NAME, NAME);
+
+    public static void register(ScalarFunctionModule scalarFunctionModule) {
+        scalarFunctionModule.register(FQN,
+                                      new BaseFunctionResolver(
+                                          FuncParams
+                                              .builder(Param.STRING)
+                                              .withVarArgs(Param.BOOLEAN)
+                                              .limitVarArgOccurrences(1)
+                                              .build()) {
+                @Override
+                public FunctionImplementation getForTypes(List<DataType> datatypes) {
+                    return new CurrentSettingFunction(
+                        new FunctionInfo(new FunctionIdent(FQN, datatypes), datatypes.get(0))
+                    );
+                }
+            });
+    }
+
+    private final FunctionInfo info;
+
+    CurrentSettingFunction(FunctionInfo info) {
+        this.info = info;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public String evaluate(TransactionContext txnCtx, Input<Object>... args) {
+        assert args.length == 1 || args.length == 2 : "number of args must be 1 or 2";
+        String settingName = (String) args[0].value();
+        SessionSetting<?> sessionSetting = SessionSettingRegistry.SETTINGS.get(settingName);
+        if (sessionSetting == null) {
+            final boolean missingOk = args.length == 2 && ((Boolean) args[1].value()) == true;
+            if (missingOk) {
+                return "";
```

Comment:
pg returns null in this case.

Reply:
ah, though the client would return NULL like crash. thanks for spotting that